pkg/genlib/generator_with_text_template.go

```diff
@@ -40,12 +40,12 @@ func NewGeneratorWithTextTemplate(tpl []byte, cfg Config, fields Fields) (*Gener
 	templateFns["generate"] = func(field string) interface{} {
 		bindF, ok := fieldMap[field]
 		if !ok {
-			return ""
+			return "<missing field>"
```

Comment:
I was thinking about this, I'd rather send a `null`, so if the value is assigned to a variable that's expected to be of another type rather than a string, and further functions are applied the template will most likely fail to render

the best would be if we could avoid any kind of failure in the `generate` body

`value, err := bindF(state, nil)` can be indeed refactored without returning any error (see [here](https://github.com/elastic/elastic-integration-corpus-generator-tool/pull/51/files#diff-0fc87380e8715fdb1a33c600e6513325ba82801e849851c5856064b7beac59b8R94)

we cannot do anything about `bindF, ok := fieldMap[field]`, but maybe let's fail the whole command, since having in a template `generate "fieldNotInFieldsYaml` is not a transient error but a bug in the template

Reply:
> I'd rather send a null

Returning `nil` does not halt execution, just renders an empty template.

> but maybe let's fail the whole command,

I tried panicking from `generate` function, but `template.Execute` recover from panics (see [here](https://github.com/golang/go/blob/0fa9ba8abeb4519db6fd5121e2794d8818daf8dc/src/text/template/exec.go#L204-L220), go1.19).  

I've not been able to have the panic triggered in `generate` to escape this recovery, due to this I've found no way to properly halt execution.

> value, err := bindF(state, nil)

I will remove the commit from this PR, we will address this in a future update.

Reply:
> I tried panicking from `generate` function, but `template.Execute` recover from panics (see [here](https://github.com/golang/go/blob/0fa9ba8abeb4519db6fd5121e2794d8818daf8dc/src/text/template/exec.go#L204-L220), go1.19).
> 
> I've not been able to have the panic triggered in `generate` to escape this recovery, due to this I've found no way to properly halt execution.

we can have an error channel and close it on `!ok`, then in the emit() we add a select with `<-errorChan` after parsing the template, that returns an error, and a default that `return nil`

Reply:
> we can have an error channel and close it on `!ok`, then in the `emit()` we add a select with `<-errorChan` after parsing the template, that returns an error, and a default that `return nil`

please, @endorama , check my proposal: https://github.com/elastic/elastic-integration-corpus-generator-tool/pull/51/commits/2e1bc5e5f0e58273288a6ed3928fe40f37809792#diff-0fc87380e8715fdb1a33c600e6513325ba82801e849851c5856064b7beac59b8R120